src/frontends/src/bench.cpp

```diff
@@ -3,21 +3,159 @@
 #include <string>
 #include <utility>
 #include <vector>
+#include <functional>
+
+#include <time.h>
 
 #include <boost/thread.hpp>
 
 #include <clipper/constants.hpp>
 #include <clipper/datatypes.hpp>
 #include <clipper/logging.hpp>
 #include <clipper/query_processor.hpp>
+#include <clipper/future.hpp>
+#include <fstream>
 
 using namespace clipper;
 
+const std::string SKLEARN_MODEL_NAME = "bench_sklearn_cifar";
+const std::string CONFIG_KEY_PATH = "path";
+const std::string CONFIG_KEY_NUM_THREADS = "num_threads";
+const std::string CONFIG_KEY_NUM_BATCHES = "num_batches";
+const std::string CONFIG_KEY_BATCH_SIZE = "batch_size";
+const std::string CONFIG_KEY_BATCH_DELAY = "batch_delay";
+
+constexpr double SKLEARN_PLANE_LABEL = 1;
+constexpr double SKLEARN_BIRD_LABEL = 0;
+
+std::unordered_map<int, std::vector<std::vector<double>>> load_cifar(
+    std::unordered_map<std::string, std::string>& config) {
+  std::ifstream cifar_file(config.find(CONFIG_KEY_PATH)->second, std::ios::binary);
+  std::istreambuf_iterator<char> cifar_data(cifar_file);
+  std::unordered_map<int, std::vector<std::vector<double>>> vecs_map;
+  for (int i = 0; i < 10000; i++) {
+    int label = static_cast<int>(*cifar_data);
+    std::cout << label << std::endl;
+    cifar_data++;
+    std::vector<uint8_t> cifar_byte_vec;
+    cifar_byte_vec.reserve(3072);
+    std::copy_n(cifar_data, 3072, std::back_inserter(cifar_byte_vec));
+    cifar_data++;
+    std::vector<double> cifar_double_vec(cifar_byte_vec.begin(), cifar_byte_vec.end());
+
+    std::unordered_map<int, std::vector<std::vector<double>>>::iterator label_vecs = vecs_map.find(label);
+    if(label_vecs != vecs_map.end()) {
+      label_vecs->second.push_back(cifar_double_vec);
+    } else {
+      std::vector<std::vector<double>> new_label_vecs;
+      new_label_vecs.push_back(cifar_double_vec);
+      vecs_map.emplace(label, new_label_vecs);
+    }
+  }
+  return vecs_map;
+}
+
+void send_predictions(std::unordered_map<std::string, std::string>& config, QueryProcessor &qp,
+                      std::unordered_map<int, std::vector<std::vector<double>>> &cifar_data,
+                      std::shared_ptr<metrics::RatioCounter> accuracy_ratio) {
+  std::vector<std::vector<double>> planes_vecs = cifar_data.find(0)->second;
```

Comment:
I get a segfault at this line. I'm guessing something is wrong with loading the cifar data?

Let me know if you can't reproduce it, and I can dig into it a little more.